packages/mask/src/extension/popups/pages/Friends/Detail/index.tsx

```diff
@@ -0,0 +1,29 @@
+import { memo, useCallback, useMemo, useState } from 'react'
+import { FriendsDetailUI } from './UI.js'
+import { useLocation } from 'react-router-dom'
+import Services from '../../../../service.js'
+import { ECKeyIdentifier } from '@masknet/shared-base'
+export const FriendsDetail = memo(() => {
+    const location = useLocation()
+    const { avatar, profiles, nextId, publicKey, isLocal } = location.state
+    const [deleted, setDeleted] = useState(false)
+    const handleDelete = useCallback(() => {
+        const personaIdentifier = ECKeyIdentifier.fromHexPublicKeyK256(nextId).expect(
+            `${nextId} should be a valid hex public key in k256`,
+        )
+        Services.Identity.deletePersona(personaIdentifier, 'safe delete')
```

Comment:
A service method always returns a promising result.

```suggestion
        await Services.Identity.deletePersona(personaIdentifier, 'safe delete')
```